scripts/monitoring/auto_scale_workers.py

```diff
@@ -0,0 +1,136 @@
+        evalai_endpoint  # Gets all challenges
+    )
+    response = execute_get_request(all_challenge_endpoint)
+
+    return response.json()
+
+
+def get_sqs_queue_by_name(queue_name):
+    """
+    Returns:
+        Returns the SQS Queue object
+    """
+
+    sqs = boto3.resource(
+        "sqs",
+        region_name=os.environ.get("AWS_DEFAULT_REGION", "us-east-1"),
+        aws_secret_access_key=os.environ.get("AWS_SECRET_ACCESS_KEY"),
+        aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),
+    )
+    if queue_name == "":
+        raise ValueError("Queue Name cannot be empty")
+
+    try:
+        queue = sqs.get_queue_by_name(QueueName=queue_name)
+    except botocore.exceptions.ClientError as ex:
+        if (
+            ex.response["Error"]["Code"]
+            != "AWS.SimpleQueueService.NonExistentQueue"
+        ):
+            raise Exception("Cannot get queue: {}".format(queue_name))
+    return queue
+
+
+def get_queue_length(queue):
+    # TODO: Check if we should use approximate number of messages
+    # https://github.com/boto/boto3/issues/599
+    # Ref: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/sqs.html#using-an-existing-queue
+    # Ref: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sqs.html#SQS.Queue.attributes
+    return (
+        queue.attributes["ApproximateNumberOfMessages"]
+        + queue.attributes["ApproximateNumberOfMessages"]
+    )
+
+
+def get_queue_length_by_challenge(challenge):
+    queue_name = challenge["queue"]
+    queue = get_sqs_queue_by_name(queue_name)
+    return get_queue_length(queue)
+
+
+def get_boto3_client(resource, aws_keys):
+    """
+    Returns the boto3 client for a resource in AWS
+    Arguments:
+        resource {str} -- Name of the resource for which client is to be created
+        aws_keys {dict} -- AWS keys which are to be used
+    Returns:
+        Boto3 client object for the resource
+    """
+    try:
+        client = boto3.client(
+            resource,
+            region_name=aws_keys["AWS_REGION"],
+            aws_access_key_id=aws_keys["AWS_ACCESS_KEY_ID"],
+            aws_secret_access_key=aws_keys["AWS_SECRET_ACCESS_KEY"],
+        )
+        return client
+    except Exception as e:
+        raise e
+
+
+def increase_or_decrease_workers(challenge):
+    queue_length = get_queue_length_by_challenge(challenge)
+    if queue_length > 0:
+        # start worker
+        start_worker(challenge["id"])
+        print("Started worker for challenge: {}".format(challenge["id"]))
+        pass
+    else:
+        # stop worker
+        stop_worker(challenge["id"])
+        print("Stopped worker for challenge: {}".format(challenge["id"]))
+        pass
+
+
+# TODO: Factor in limits for the APIs
+def increase_or_decrease_workers_for_challenges(response):
+    for challenge in response["results"]:
+        if not challenge["use_host_credentials"]:
```

Comment:
Can you add another check here - also check if `remote_evaluation` is not set to `True` for a challenge. We only want to run auto-scaling for workers at our end.